### DISCUSSION TITLE
Upcoming Feature Planning

### DISCUSSION THREAD

**Discussion by 1hitsong** [jellyfin-archive/jellyfin-roku-legacy#662] [Ideas]
What new features, fixes, changes, etc. should we include as part of 1.4.15? Currently, this release is the one after the next, so this is future planning and brainstorming.

Think in terms of "must haves," "nice to haves," and "stretch goals."

### Must Haves
- [x] User Setting: Hide clock in header #697 
- [x] Music: Add Genre View #881 
- [x] Voice Search #593 
- [x] Movies: Local Trailer Support #761 
- [x] User Setting: Go directly to episode list if only 1 season loaded #919 
- [x] Review support for Max Days in Next Up setting #767 
- [x] Add ReportPlayback() functionality to Music #768 
- [ ] Expand and fill in View, Sort, Filter options in library grids to match Web as much as possible #928 
- [x] ~~Support for Intro-Skipper plugin #696~~
- [x] Removing book listings code #639 
- [x] Basic Playlist Support (MVP) #638 
- [ ] Live TV Guide: View, Sort, Filter options
- [ ] Live TV: Create mini guide while watching
- [ ] Live TV: Add restart/reload functionality on buffering error
- [x] Music: Improve design of Album screen

### Nice To Haves
- [x] Add setting: User defined load item count #655 
- [x] TV Shows: Season Shuffle Play
- [ ] Movies: Play Random

### Stretch Goals
- [ ] Add Instant Resume
- [ ] Add OSD (On Screen Display) options for the video types listed below

**Comment by candry7731** [comment]
I would like to expand the View, Sort, Filter options for the library grid. 
Add more functionally to the Live TV Guide -  View, Sort, Filter options
Add a mini guide menu while watching Live TV
Add the ability to restart/reload a LiveTV stream if there is a buffering error (The android/Firestick client does this)
Add Instant Resume -  However this is over my head....

**Reply by lakerssuperman** [reply]
Humble user here again.  I don't know how many pieces have to fall into place to make this happen, but sorting of boxsets/collections of movies by release order would be awesome to have as it would match the web interface default and add a little more polish.  There are several open issues about it.  https://github.com/jellyfin/jellyfin-roku/issues/391

Again, just a suggestion for consideration.  You guys are doing awesome work and I very much appreciate it.  Thank you.

**Reply by candry7731** [reply]
I will test this tonight. It looks like the PR #393 should have added this functionality as it added PremiereDate to the sort option. Does this work? 

If it does work, we can add this to the Roku User Preferences in the settings menu. We could have 2 options in the settings menu. 1 option to set "all folders" sort default and a second option to set "Collections / Boxset" sort default.

**Reply by lakerssuperman** [reply]
From what I I've seen in my testing, it allows the entire "Collections" section to be sorted by release date.  However, inside individual collections, for example my James Bond movies, the movies are still sorted in alphabetical order and the release date sort option is not available.

**Reply by candry7731** [reply]
Yea, Sorting the entire collections folder by release date does Not really make sense in my opinion. 

I will look at this tonight and see if I can get a PR together should be a easy update :)

**Reply by lakerssuperman** [reply]
That'd be awesome!  Thank you!!

**Reply by candry7731** [reply]
Done! #663 should add the "ReleaseDate" option to Boxsets. When the Item Count PR is released we will build out the user setting like I mentioned

**Reply by lakerssuperman** [reply]
Wow!  Thank you so much!!  Awesome stuff.

**Comment by candry7731** [comment]
Stretch Goal: Add OSD (On Screen Display) options for the following video types

LiveTV -
- [ ] Program Info
- [ ] Mini Guide
- [ ] Record button/Icon
- [ ] favorite button/Icon

Movies OSD
- [ ] Movie Info 
- [ ] Cast & Crew
- [ ] Chapters
- [ ] Movie Suggestions
- [ ] Favorite button/Icon

Episode OSD
- [ ] Episode Info
- [ ] Cast & Crew
- [ ] Episode List
- [ ] Favorite button/Icon

Inspiration:
![image](https://user-images.githubusercontent.com/83685828/174696867-3b1a0eda-327f-4d97-9c7b-799ebb4a3bbf.png)

**Reply by hoboken411** [reply]
One other piece of data missing (that is available on the Android TV version), is the TIME the movie ends when you hit pause.
The Android version shows what time the movie is projected to end when you pause the movie. And it continually updates as the movie remains paused.

**Comment by p37307** [comment]
I know this post is from last year, but since it is still open, I would like to opine in.

Since the home screen only shows a limited amount of "Next Up', and profile display settings lets decide how long a _show should stay in the 'Next Up' list without watching it_, it would be great if there were a way to get to a screen to view all the 'Next up' items.

On the web, you can click Next up's header on the home screen and it takes you to an expanded list/grid. The web URL to see what I am talking about is localhost:8096/web/index.html#!/list.html?type=nextup&serverId=xxxxxxxxxxx

It's common to start a series and move on but want to go back and restart it at a later date and the 15 or 24, however many is set to appear in the home screen just isn't enough.

Another alternative would be to let the user set the number of items from a settings option in the Roku App that says how many items appear in the home screen's Next Up row. I know there is a setting to set the number of days a show should remain in next up, just not anything to show how to display them. Not sure if that is _getNextUpEpisodes({Limit:24_ or not and how easy that would be to put in a settings config.

If I missed a way to do what I am proposing, please let me know. I know we can sort Libraries by 'date played' but then you have to open the library and click into the show and then season until you get to the next up episode.

**Reply by 1hitsong** [reply]
I like the idea of adding a "View All" next up view. Go ahead and submit it as a feature request so it's properly tracked and not lost here in the discussions, which are rarely checked.

https://github.com/jellyfin/jellyfin-roku/issues/new?assignees=&labels=feature&projects=&template=feature_request.md&title=

**Reply by p37307** [reply]
Thanks, and I completed it here: https://github.com/jellyfin/jellyfin-roku/issues/1481

**Reply by cewert** [reply]
iirc the limit of 16 is because "that's what the web client does". I don't believe this number was chosen based on testing or device limitations. It would be easy to increase this number or tie it to a user setting but having a view where you can see them all would probably be best. Maybe both (the last item in the list could link to the "view all" page)? We've been defaulting to web client settings lately with the option to override them in our apps user settings.

**Comment by p37307** [comment]
> (the last item in the list could link to the "view all" page)? 

I like this option, even if you give the option to override the number in the settings. We use multiple Roku's in our home so it would be a uniform experience.

**Comment by Alliyance** [comment]
Fix the subs #1665. 
Been 10 months

**Reply by jimdogx** [reply]
> Fix the subs #1665. Been 10 months

https://github.com/jellyfin/jellyfin-roku/pull/2020

**Reply by Alliyance** [reply]
Wow. Thank you to fix that up!
